airqo-platform/AirQo-api
# Ft caching gp

# Decreasing response time for GP model predictions

**_WHAT DOES THIS PR DO?_**
This PR introduces caching for the predict heatmap endpoint as well as saving predictions in a database table.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**

**_WHAT IS THE LINK TO THE PR BRANCH_**

**_HOW DO I TEST OUT THIS PR?_**
Git clone this repo and then run the commands below:
`cd src/predict`
`env\scripts\activate` for windows or `source env\bin\activate` for MAC OS
`pip install -r requirements.txt`
`flask run`

Note: Please make sure redis is running.

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
`http://localhost:5000/api/v1/predict/heatmap`

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**
